File: Whatsapp/Message.m
//
//  Message.m
//  Whatsapp
//
//  Created by Rafael Castro on 6/16/15.
//  Copyright (c) 2015 HummingBird. All rights reserved.
//

#import "Message.h"

@implementation Message

-(id)init
{
    self = [super init];
    if (self)
    {
        self.sender = MessageSenderMyself;
        self.status = MessageStatusSending;
        self.text = @"";
        self.heigh = 44;
        self.date = [NSDate date];
        self.identifier = @"";
    }
    return self;
}

+(Message *)messageFromDictionary:(NSDictionary *)dictionary
{
    Message *message = [[Message alloc] init];
    message.text = dictionary[@"text"];
    message.identifier = dictionary[@"message_id"];
    message.status = [dictionary[@"status"] integerValue] + 1;
    
    NSString *dateFormat = @"yyyy-MM-dd'T'HH:mm:ss.SSSSSS";
    
    //Date in UTC
    NSTimeZone *inputTimeZone = [NSTimeZone timeZoneWithAbbreviation:@"UTC"];
    NSDateFormatter *inputDateFormatter = [[NSDateFormatter alloc] init];
    [inputDateFormatter setTimeZone:inputTimeZone];
    [inputDateFormatter setDateFormat:dateFormat];
    NSDate *date = [inputDateFormatter dateFromString:dictionary[@"sent"]];
    
    //Convert time in UTC to Local TimeZone
    NSTimeZone *outputTimeZone = [NSTimeZone localTimeZone];
    NSDateFormatter *outputDateFormatter = [[NSDateFormatter alloc] init];
    [outputDateFormatter setTimeZone:outputTimeZone];
    [outputDateFormatter setDateFormat:dateFormat];
    NSString *outputString = [outputDateFormatter stringFromDate:date];
    
    message.date = [outputDateFormatter dateFromString:outputString];
    
    return message;
}
@end

File: Whatsapp-Swift/Whatsapp-Swift/Message.swift
//
//  Message.swift
//  Whatsapp
//
//  Created by Rafael Castro on 6/16/15. / Swift Version by Breno Oliveira on 02/18/16.
//  Copyright (c) 2015 HummingBird. All rights reserved.
//

import UIKit

enum MessageStatus {
    case Sending
    case Sent
    case Received
    case Read
    case Failed
}

enum MessageSender {
    case Myself
    case Someone
}

class Message:NSObject {
    var sender:MessageSender = .Myself
    var status:MessageStatus = .Sending
    var identifier:String = ""
    var chatId:String = ""
    var text:String = ""
    var date:NSDate = NSDate()
    var height:CGFloat = 44

    func messageFromDictionary(dictionary:NSDictionary) -> Message {
        let message = Message()
        message.text = dictionary["text"] as! String
        message.identifier = dictionary["messageId"] as! String
        message.status = dictionary["status"] as! MessageStatus
        
        let dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSSSSS"
        
        //Date in UTC
        let inputTimeZone = NSTimeZone(abbreviation: "UTC")
        let inputDateFormatter = NSDateFormatter()
        inputDateFormatter.timeZone = inputTimeZone
        inputDateFormatter.dateFormat = dateFormat
        let date = inputDateFormatter.dateFromString(dictionary["sent"] as! String)
        
        //Convert time in UTC to Local TimeZone
        let outputTimeZone = NSTimeZone.localTimeZone()
        let outputDateFormatter = NSDateFormatter()
        outputDateFormatter.timeZone = outputTimeZone
        outputDateFormatter.dateFormat = dateFormat
        let outputString = outputDateFormatter.stringFromDate(date!)
        
        message.date = outputDateFormatter.dateFromString(outputString)!
        
        return message
    }
}